### ISSUE TITLE
False positives for SA1009 and SA1011

### ISSUE THREAD

**Issue by ArthurHNL** [DotNetAnalyzers/StyleCopAnalyzers#3143]
When writing a value converter for SQLite Concurrency Tokens in EF Core, I discovered two false positives:

```csharp
    public class SqliteConcurrencyTokenConverter : ValueConverter<byte[], string>
    {
        public SqliteConcurrencyTokenConverter()
            : base(
                  (v) => ToDb(v)!, // SA1009: Closing parenthesis should be followed by a space.
                  (v) => FromDb(v)!)
        {
        }

        private static byte[]? FromDb(string? v) => // SA1011: Closing square bracket should be followed by a space.
            v != null ? Encoding.UTF8.GetBytes(v) : null;

        private static string? ToDb(byte[]? v) =>
            v != null ? Encodin
```

The first false positive (SA1009) is triggered by the null forgiving operator (!) directly after the closing parenthesis of the `ToDb` call. The second false positive (SA1011) is caused by having a nullable type symbol placed after the closing square bracket of an array type.

**Comment by sharwell** [comment]
Both of these issues appear to already be fixed. I'm adding a test case for SA1009. The SA1011 issue is already covered by #2927.

**Comment by abbotware** [comment]
Just came across the SA1011 issue myself
